lib/util.ts

```diff
@@ -47,51 +53,31 @@ exports.strToChars = (str) => {
  * @param {string} regexpStr
  * @return {Array.<Array.<Object>, number>}
  */
-exports.tokenizeClass = (str, regexpStr) => {
-  /* jshint maxlen: false */
-  let tokens = [];
+export const tokenizeClass = (str: string, regexpStr: string): [SetTokens, number] => {
+  let tokens: SetTokens = [], rs: string[] | null, c: string;
   const regexp = /\\(?:(w)|(d)|(s)|(W)|(D)|(S))|((?:(?:\\)(.)|([^\]\\]))-(?:\\)?([^\]]))|(\])|(?:\\)?([^])/g;
-  let rs, c;
-
 
   while ((rs = regexp.exec(str)) != null) {
-    if (rs[1]) {
-      tokens.push(sets.words());
-
-    } else if (rs[2]) {
-      tokens.push(sets.ints());
-
-    } else if (rs[3]) {
-      tokens.push(sets.whitespace());
-
-    } else if (rs[4]) {
-      tokens.push(sets.notWords());
-
-    } else if (rs[5]) {
-      tokens.push(sets.notInts());
-
-    } else if (rs[6]) {
-      tokens.push(sets.notWhitespace());
-
-    } else if (rs[7]) {
-      tokens.push({
+    const p = (rs[1] && sets.words())
+      ?? (rs[2] && sets.ints())
+      ?? (rs[3] && sets.whitespace())
+      ?? (rs[4] && sets.notWords())
+      ?? (rs[5] && sets.notInts())
+      ?? (rs[6] && sets.notWhitespace())
+      ?? (rs[7] && {
         type: types.RANGE,
         from: (rs[8] || rs[9]).charCodeAt(0),
-        to: rs[10].charCodeAt(0),
-      });
+        to: rs[10].charCodeAt(0)
+      })
+      ?? ((c = rs[12]) && { type: types.CHAR, value: c.charCodeAt(0) })
 
-    } else if ((c = rs[12])) {
-      tokens.push({
-        type: types.CHAR,
-        value: c.charCodeAt(0),
-      });
-
-    } else {
+    if (p) 
+      tokens.push(p);
+    else 
       return [tokens, regexp.lastIndex];
-    }
-  }
-
-  exports.error(regexpStr, 'Unterminated character class');
+  };
+  error(regexpStr, 'Unterminated character class')
+  throw new SyntaxError(`Invalid regular expression: ${str}: Unterminated character class`)
```

Comment:
this is never reached since the line above throws

Reply:
The typescript compiler doesn't recognise this. When you try to compile without line 80 it returns `TS7030: Not all code paths return a value.`

My reccomendation would be to do away with the error utility and just have 
```
throw new SyntaxError('Invalid regular expression: /' + regexpStr + '/: ');
```
in line 79

Reply:
> My reccomendation would be to do away with the error utility

👍